index/find_component.go

```diff
@@ -189,6 +189,11 @@ func (index *SpecIndex) lookupRemoteReference(ref string) (*yaml.Node, *yaml.Nod
     } else {
         query = "$"
     }
+    
+    query, err := url.PathUnescape(query)
```

Comment:
Nice upgrade!

My only comment is that this branch will drop code coverage without a test to check for the invalid case.

Reply:
thank you. Sure I am going to add some testing